src/bos/operators/utils/boot_image_metadata/s3_boot_image_metadata.py

```diff
@@ -204,3 +205,41 @@ def manifest_s3_url(self) -> S3Url:
         Returns the S3 URL to the boot manifest
         """
         return self.boot_artifacts.s3url
+
+    @property
+    def arch(self):
+        """
+        Extract the IMS image ID from the S3 manifest path.
+        Query IMS to get the architecture of the image.
+        Return the 'arch' field from the IMS image
+        If image is not in IMS, or 'arch' field not set, log warnings and return None.
+        (since technically BOS does not require the images to be in IMS)
+        """
+        s3_url = self.manifest_s3_url
+        ims_id = get_ims_id_from_s3_url(s3_url)
+        if not ims_id:
+            LOGGER.warning(
+                "Boot artifact S3 URL '%s' does not follow the expected IMS image convention",
+                s3_url)
+            return None
+        try:
+            ims_image_data = get_image(ims_id)
+        except ImageNotFound:
+            LOGGER.warning(
+                "Can't determine architecture of '%s' because image '%s' does not exist in IMS",
+                s3_url.url, ims_id)
+            return None
+        except Exception as err:
+            LOGGER.error("Error getting IMS image data for '%s' (S3 path '%s'): %s", ims_id,
+                         s3_url.url, exc_type_msg(err))
+            return None
+        try:
+            return ims_image_data["arch"]
+        except KeyError:
+            LOGGER.warning("Can't determine architecture of '%s' because 'arch' field not set in "
+                           "IMS image '%s': %s", s3_url.url, ims_id, ims_image_data)
+        except Exception as err:
+            LOGGER.error("IMS image '%s' (s3 path '%s'): %s", ims_id, s3_url.url, ims_image_data)
+            LOGGER.error("Error getting 'arch' field for IMS image '%s': %s", ims_id,
+                         exc_type_msg(err))
+        return None
```

Comment:
By convention, we typically define no stored arch as x86 for reasons of backwards compat.

Reply:
Yeah, I figured that. I wasn't sure if we wanted to do that here because per the IMS spec, the `arch` field is required, so it really should never be absent in CSM 1.6. But I guess I could still log the error but treat it as X86.

Reply:
Oh, I take it back -- I must have dreamed that it was required for IMS. I'll update this to default accordingly.

Reply:
Good catch!